### DISCUSSION TITLE
Testing with jest and jest-environment-jsdom using undici - error:  ReferenceError: ReadableStream is not defined

### DISCUSSION THREAD

**Discussion by Roman-e-prog** [mswjs/msw#1925] [Q&A]
I have installed msw 2.0.2 into a next.js 13.4.10 project that will be tested with jest. I had several errors and followed the instructions from the msw docs. There I was told to build a jest.polyfills.js in the root of the project and that I should install undici. This I have done and coppied the code from the docs. Now I have this error when running a jest test. ReadableStream is not defined. I am not familiar with this undici library, not used polyfills until yet and just starting with msw. So I not know how I should deal with this. I don't want to build a custom test environment. Adding ReadableStream to the polyfill like this not worked: 
```
import { ReadableStream } from 'web-streams-polyfill/ponyfill'
global.ReadableStream = ReadableStream
```
Is there another solution from msw? Perhaps someone can advice me? 

Current error:
Test suite failed to run

    ReferenceError: ReadableStream is not defined

      19 | const { Blob } = require('node:buffer')
      20 | const { fetch, Headers, FormData, Request, Response } = require('undici')
    > 21 |
         | ^
      22 | Object.defineProperties(globalThis, {
      23 |   fetch: { value: fetch, writable: true },
      24 |   Blob: { value: Blob },

      at Object.<anonymous> (node_modules/undici/lib/fetch/response.js:506:3)       
      at Object.<anonymous> (node_modules/undici/index.js:118:27)
      at Object.<anonymous> (jest.polyfills.js:21:57)

**Comment by mattcosta7** [comment]
I think there's a change in unidic v6 that we haven't dug into yet, try with v5 instead and let us know if that doesn't help?

**Reply by mattcosta7** [reply]
https://github.com/mswjs/msw/issues/1916#issuecomment-1845769931

this also should work, but the order matters (ReadableStream has to be set earlier than unidici's import

**Comment by devamitranjan** [comment]
const { TextDecoder, TextEncoder, ReadableStream } = require("node:util");
This worked for me.
